src/isomorphic/hooks/ReactComponentTreeHook.js

```diff
@@ -402,6 +402,57 @@ var ReactComponentTreeHook = {
 
   getRootIDs,
   getRegisteredIDs: getItemIDs,
+
+  pushNonStandardWarningStack(
+    isCreatingElement: boolean,
+    currentSource: ?Source,
+  ) {
+    if (typeof console.stack !== 'function') {
+      return;
+    }
+
+    var stack = [];
+    var currentOwner = ReactCurrentOwner.current;
+    var id = currentOwner && currentOwner._debugID;
+
+    try {
+      if (isCreatingElement) {
+        stack.push({
+          functionName: id ? ReactComponentTreeHook.getDisplayName(id) : null,
```

Comment:
Let's call this just `name`. It could be a class constructor, method, native method, a "message" or whatever else you might call this thing.